peps/pep-0762.rst

```diff
@@ -0,0 +1,276 @@
+and is available on demand, in case custom workflows require it. It will also be
+used in cases where the new REPL cannot be used due to environmental constraints
+or other issues.  Users have the option to explicitly choose the old basic REPL
+by setting the environment variable ``PYTHON_BASIC_REPL`` to 1. This ensures
+that users can continue using the familiar interface and capabilities if they
+prefer, or if they encounter any issues with the new implementation.
+
+It's important to emphasize that the introduction of PyREPL does not remove any
+existing functionality. Any functionality of the old basic REPL unavailable in
+PyREPL is preserved and maintained in the old basic REPL that can be used by
+users as a fallback.
+
+In particular, users wanting to continue using their custom input configuration
+in ``inputrc`` or ``editrc`` files, can continue using the old basic REPL.
+
+The authors do not expect any PyREPL functionality to be ported to the old basic
+REPL. Similarly, ``inputrc`` and ``editrc`` support is explicitly not planned in
+PyREPL. Those configuration files are provided by and parsed by “readline” and
+“editline” libraries, and their functional scope does not match the
+functionality PyREPL is targeting.
+
+To facilitate a smooth transition, clear documentation is provided on how to
+switch between PyREPL and the old basic REPL.
+
+This approach ensures that while we're introducing significant improvements with
+the new REPL, we're not forcing any immediate changes on users who rely on the
+current implementation. The fallback mechanism and user choice option provide a
+safety net that allows for gradual adoption of the new REPL while maintaining
+all existing functionality.
+
+Security Implications
+=====================
+
+There are no security implications derived from this proposal.
+
+How to Teach This
+=================
+
+TODO: Make these points more concrete???
+
+The introduction of  PyREPL is accompanied by documentation and tutorials. Key
+areas of focus for education will include:
+
+1. Detailed explanations on using multi-line editing, paste mode, and other new
+   features.
+
+2. What custom commands are available (existing and new).
+
+3. Explanations on how to switch from the old to the new REPL, including any
+   differences from the previous readline/editline-based configuration.
+
+Rejected Ideas
+==============
+
+Several alternative approaches were considered and ultimately rejected:
+
+1. Extending the current C Implementation: While this would maintain maximum
+   backwards compatibility, it was deemed too complex and would not address the
+   fundamental limitations described ut supra.
+
+2. Developing a new REPL from scratch: This approach was rejected due to the
+   complexity of creating a cross-platform terminal application and the desire to
+   leverage existing, proven code.
+
+3. Using Other Existing REPL Implementations: While other Python REPL
```

Comment:
It probably makes sense to address why IPython was not chosen as it has better name recognition and adoption than PyREPL.